src/main/java/com/google/songgame/data/UserCookieReader.java

```diff
@@ -0,0 +1,30 @@
+package com.google.songgame.data;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import javax.servlet.http.Cookie;
+
+/**
+ * Helper class to read data from Users Cookies
+ */
+public final class UserCookieReader {
+  /**
+   * Helper method that reads and returns the user id stored in a cookie
+   */
+  public static String getUserId(HttpServletRequest request) throws IOException {
```

Comment:
Could you add some tests for this method? The easiest way to do so would be to refactor it to take a Cookie[] array as a parameter. I checked the docs and Cookie has a constructor that takes two strings as parameters, so it should be easy to construct example Cookie objects in the test:
https://docs.oracle.com/javaee/6/api/javax/servlet/http/Cookie.html

Reply:
Otherwise, this is looking great